src/java/htsjdk/samtools/CRAMFileWriter.java

```diff
@@ -134,10 +149,6 @@ private void initCRAMWriter(final OutputStream indexOS, final ReferenceSource so
         setSortOrder(samFileHeader.getSortOrder(), preSorted);
         setHeader(samFileHeader);
 
-        if (this.source == null) {
-            this.source = new ReferenceSource(Defaults.REFERENCE_FASTA);
-        }
```

Comment:
If I understand the question, yes. The consumer should create a reference source by whatever mechanism it wants, perhaps by calling getDefaultReferenceSource, and pass it in to one of these constructors. The whole idea of this PR was to have code at this level (CRAMFileReader and CRAMFileWriter) always require that a reference be provided, and give the higher levels (i.e. SAMFileWriterFactory) a unified way to create a default to provide to the lower levels via getDefaultReferenceSource. Not sure if that answers the question.